Sources/MessagingInApp/Views/UIKitInline.swift

```diff
@@ -206,50 +167,25 @@ public class InAppMessageView: UIView {
-        newInlineMessageManager.delegate = Gist.shared
-
-        let inlineView = newInlineMessageManager.inlineMessageView
-        inlineView.isHidden = true // start hidden while the message renders. When complete, it will show the View.
-
-        addSubview(inlineView)
-
-        // Setup the WebView to be the same size as this View. When this View changes size, the WebView will change, too.
-        inlineView.translatesAutoresizingMaskIntoConstraints = false // Required in order for this inline View to have full control over the AutoLayout constraints for the WebView.
-        NSLayoutConstraint.activate([
-            inlineView.topAnchor.constraint(equalTo: topAnchor),
-            inlineView.leadingAnchor.constraint(equalTo: leadingAnchor),
-            inlineView.trailingAnchor.constraint(equalTo: trailingAnchor),
-            inlineView.bottomAnchor.constraint(equalTo: bottomAnchor)
-        ])
-
-        inlineMessageManager = newInlineMessageManager
-    }
-
-    private func stopShowingMessageAndCleanup() {
-        // If a message is currently being shown, cleanup and remove the webview so we can begin showing a new message.
-        // Cleanup needs to involve removing the WebView from it's superview and cleaning up the WebView's resources.
-        inlineMessageManager?.stopAndCleanup()
-        inlineMessageManager?.inlineMessageView.removeFromSuperview()
-        inlineMessageManager = nil
-    }
+    private func animateHeight(to height: CGFloat) {
+        // this function can be called multiple times in short period of time so we could be in the middle of 1 animation. Cancel the current one and start new.
+        runningHeightChangeAnimation?.stopAnimation(true)
 
-    private func dismissInAppMessage() {
-        // If this function gets called a lot in a short amount of time (eventbus triggers multiple events), the dismiss animation does not look as expected.
-        // To fix this, exit early if dismiss has already been triggered.
-        if inlineMessageManager?.inlineMessageDelegate == nil {
-            return
-        }
+        runningHeightChangeAnimation = UIViewPropertyAnimator(duration: 0.3, curve: .easeIn, animations: {
+            self.heightConstraint?.constant = height // Changing the height in animation block indicates we want to animate the height change.
 
-        stopShowingMessageAndCleanup()
+            // Since we modified constraint, perform a UI refresh to apply the change.
+            // It's important that we call layoutIfNeeded on the topmost superview in the hierarchy. During development, there were animiation issues if layoutIfNeeded was called on a different superview then the root.
+            // Example, given this UI:
+            // UIViewController
+            // └── UIStackView
+            //    └── InAppMessageView
+            // ...If we call layoutIfNeeded on superview (UIStackView), the animation will not work as expected.
```

Comment:
Do we expect any changes from the user to help animation work properly ?

Reply:
No, I do not expect any changes need to be made.